### PR TITLE
Link resources when revision is present.

### LINKED ISSUE #2660: Resources section in Charm details page don't have clickable links
The "resources" section in a charm details page on `jujucharms.com` has clickable links to download the resources. The Juju GUI and JaaS list the resources but without clickable links.

Example Charm: https://jujucharms.com/charmscaler/0

Resources section of details:

image

### PR DESCRIPTION
**Pull request by kadams54**
Fixes #2660.

### CODE REVIEW

**Comment by kadams54** [comment]
## QA

1. Fire up `make run` + guiproxy.
2. Search the store for "charmscaler".
3. On the charm details page, verify that the docker image in the resources section is linked.
4. Verify that the link works.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7013/

File: jujugui/static/gui/src/app/components/entity-details/content/resources/resources.js [hunk 1]
```diff
@@ -23,39 +23,77 @@ YUI.add('entity-resources', function() {
...
+      const revision = resource.Revision;
+      const name = resource.Name;
+      // Get the file extension.
+      const parts = resource.Path.split('.');
+      let extension = '';
+      // If there is a file extension then format it for display.
+      if (parts.length > 1) {
+        extension = `(.${parts.pop()})`;
+      }
+      let itemContent = null;
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/entity-details/content/resources/resources.js]: `itemContent` can be set to the 'else' value of it and then drop the else.

File: jujugui/static/gui/src/app/components/entity-details/content/resources/resources.js [hunk 2]
```diff
@@ -23,39 +23,77 @@ YUI.add('entity-resources', function() {
...
+      // A revision of 0 should not be considered false.
+      if (revision !== undefined) {
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/entity-details/content/resources/resources.js]: Why does having a revision mean it's downloadable?  @fabricematrat you had originally implemented this logic in bb

File: jujugui/static/gui/src/app/components/entity-details/content/resources/resources.js [hunk 3]
```diff
@@ -23,39 +23,77 @@ YUI.add('entity-resources', function() {
...
+      apiUrl: React.PropTypes.string.isRequired,
+      entityId: React.PropTypes.string.isRequired,
       pluralize: React.PropTypes.func.isRequired,
       resources: React.PropTypes.array
     },
 
+    getInitialState: function() {
+      const props = this.props;
+      return {
+        baseUrl: `${props.apiUrl}/${props.entityId}/resource`
```
Comment (frankban) [inline jujugui/static/gui/src/app/components/entity-details/content/resources/resources.js]: Should this use `jujulib.URL.fromLegacyString(props.entityId).legacyPath()` instead? That way we are being more explicit about what we are expecting and also we avoid including the cs: schema in the resulting URL. But maybe I am missing something and the cs: schema is never there.

File: jujugui/static/gui/src/app/components/entity-details/content/resources/resources.js (hunk 2 above)
Comment (fabricematrat) [inline jujugui/static/gui/src/app/components/entity-details/content/resources/resources.js]: You can create a charm with resource but not attach the content yet, and I think you get -1 as a revision in that case. As soon as a resource is attached it gets a revision and is downloadable. 
That if I remember correctly.

Comment (hatched) [inline jujugui/static/gui/src/app/components/entity-details/content/resources/resources.js]: @kadams54 can you update this code to include this comment plz.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7029/

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7030/

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7031/

**Comment by frankban** [comment]
QA ok.
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Comment by jujugui** [comment]
Build failed: Attempt to land pull request failed
        build url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge/2221

**Comment by frankban** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**